ddtrace/tracer/tracer.go

```diff
@@ -55,6 +55,9 @@ type tracer struct {
 
 	// pid of the process
 	pid string
+
+	// statsd client for tracer metrics.
+	statsd *statsd.Client
```

Comment:
If this were a `statsd.ClientInterface`, you'd be able to use the `NoOpClient` and avoid all the nil checks.

Reply:
I think it's a good idea to use an unexported interface here, `statsdClient` which only contains the methods that we're using, e.g. `Incr` and `Close` and use that everywhere. By default we'd have a no-op one.